Hi am still getting a crash on adding bracket view to the fragment, i have just added in the layout and the app crashes as soon as the fragment is called.I have added the log for reference and i am using the updated version. Thanks in advance,

Process: com.halil.ozel.unsplashexample, PID: 18477
                                                                                                    android.view.InflateException: Binary XML file line #19 in com.halil.ozel.unsplashexample:layout/fragment_brackets: Binary XML file line #19 in com.halil.ozel.unsplashexample:layout/fragment_brackets: Error inflating class com.ventura.bracketslib.BracketsView
                                                                                                    Caused by: android.view.InflateException: Binary XML file line #19 in com.halil.ozel.unsplashexample:layout/fragment_brackets: Error inflating class com.ventura.bracketslib.BracketsView
                                                                                                    Caused by: java.lang.reflect.InvocationTargetException
                                                                                                    	at java.lang.reflect.Constructor.newInstance0(Native Method)
                                                                                                    	at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
                                                                                                    	at android.view.LayoutInflater.createView(LayoutInflater.java:858)
                                                                                                    	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:1010)
                                                                                                    	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:965)
                                                                                                    	at android.view.LayoutInflater.rInflate(LayoutInflater.java:1127)
                                                                                                    	at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:1088)
                                                                                                    	at android.view.LayoutInflater.inflate(LayoutInflater.java:686)
                                                                                                    	at android.view.LayoutInflater.inflate(LayoutInflater.java:538)
                                                                                                    	at com.halil.ozel.unsplashexample.databinding.FragmentBracketsBinding.inflate(FragmentBracketsBinding.java:45)
                                                                                                    	at com.halil.ozel.unsplashexample.databinding.FragmentBracketsBinding.inflate(FragmentBracketsBinding.java:39)
                                                                                                    	at com.halil.ozel.unsplashexample.ui.fragment.brackets.BracketsFragment.onCreateView(BracketsFragment.kt:32)
                                                                                                    	at androidx.fragment.app.Fragment.performCreateView(Fragment.java:3104)
                                                                                                    	at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:524)
                                                                                                    	at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:261)
                                                                                                    	at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:1899)
                                                                                                    	at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:1817)
                                                                                                    	at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1760)
                                                                                                    	at androidx.fragment.app.FragmentManager$5.run(FragmentManager.java:547)
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:938)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:226)
                                                                                                    	at android.os.Looper.loop(Looper.java:313)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8663)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)
                                                                                                    Caused by: java.lang.ClassCastException: dagger.hilt.android.internal.managers.ViewComponentManager$FragmentContextWrapper cannot be cast to androidx.fragment.app.FragmentActivity
                                                                                                    	at com.ventura.bracketslib.BracketsView.<init>(BracketsView.java:45)
                                                                                                    	at java.lang.reflect.Constructor.newInstance0(Native Method) 
                                                                                                    	at java.lang.reflect.Constructor.newInstance(Constructor.java:343) 
                                                                                                    	at android.view.LayoutInflater.createView(LayoutInflater.java:858) 
                                                                                                    	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:1010) 
                                                                                                    	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:965) 
                                                                                                    	at android.view.LayoutInflater.rInflate(LayoutInflater.java:1127) 
                                                                                                    	at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:1088) 
                                                                                                    	at android.view.LayoutInflater.inflate(LayoutInflater.java:686) 
                                                                                                    	at android.view.LayoutInflater.inflate(LayoutInflater.java:538) 
                                                                                                    	at com.halil.ozel.unsplashexample.databinding.FragmentBracketsBinding.inflate(FragmentBracketsBinding.java:45) 
                                                                                                    	at com.halil.ozel.unsplashexample.databinding.FragmentBracketsBinding.inflate(FragmentBracketsBinding.java:39) 
                                                                                                    	at com.halil.ozel.unsplashexample.ui.fragment.brackets.BracketsFragment.onCreateView(BracketsFragment.kt:32) 
                                                                                                    	at androidx.fragment.app.Fragment.performCreateView(Fragment.java:3104) 
                                                                                                    	at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:524) 
                                                                                                    	at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:261) 
                                                                                                    	at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:1899) 
                                                                                                    	at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:1817) 
                                                                                                    	at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1760) 
                                                                                                    	at androidx.fragment.app.FragmentManager$5.run(FragmentManager.java:547) 
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:938) 
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:99) 
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:226) 
                                                                                                    	at android.os.Looper.loop(Looper.java:313) 
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8663) 
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method) 
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571) 
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135) 

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -44,6 +44,8 @@ dependencies {
     implementation 'com.squareup.retrofit2:converter-gson:2.9.0'
     implementation 'com.squareup.okhttp3:logging-interceptor:4.11.0'
 
+    implementation 'com.github.bumptech.glide:glide:4.15.1'
+
     testImplementation 'junit:junit:4.13.2'
     androidTestImplementation 'androidx.test.ext:junit:1.1.5'
     androidTestImplementation 'androidx.test.espresso:espresso-core:3.5.1'

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -32,7 +32,7 @@ android {
 }
 
 dependencies {
-    implementation 'com.github.emil-ep:TournamentBracketLib:1.1.1'
+    implementation 'com.github.emil-ep:TournamentBracketLib:1.1.4'
 
     implementation 'androidx.appcompat:appcompat:1.6.1'
     implementation 'com.google.android.material:material:1.9.0'

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -32,13 +32,17 @@ android {
 }
 
 dependencies {
-    implementation 'com.github.emil-ep:TournamentBracketLib:1.1.0'
+    implementation 'com.github.emil-ep:TournamentBracketLib:1.1.1'
 
     implementation 'androidx.appcompat:appcompat:1.6.1'
     implementation 'com.google.android.material:material:1.9.0'
     implementation 'androidx.constraintlayout:constraintlayout:2.1.4'
     implementation 'androidx.recyclerview:recyclerview:1.3.0'
     implementation 'androidx.viewpager2:viewpager2:1.0.0'
+
+    implementation 'com.squareup.retrofit2:retrofit:2.9.0'
+    implementation 'com.squareup.retrofit2:converter-gson:2.9.0'
+    implementation 'com.squareup.okhttp3:logging-interceptor:4.11.0'
 
     testImplementation 'junit:junit:4.13.2'
     androidTestImplementation 'androidx.test.ext:junit:1.1.5'

--- a/app/src/main/AndroidManifest.xml
+++ b/app/src/main/AndroidManifest.xml
@@ -2,6 +2,8 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="com.ventura.emilp.tournamentbrackets">
 
+    <uses-permission android:name="android.permission.INTERNET" />
+
     <application
         android:allowBackup="true"
         android:icon="@mipmap/ic_launcher"

--- a/app/src/main/AndroidManifest.xml
+++ b/app/src/main/AndroidManifest.xml
@@ -11,14 +11,24 @@
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
         android:theme="@style/AppTheme">
-        <activity android:name=".activity.MainActivity"
+
+        <activity
+            android:name=".activity.HomeActivity"
             android:exported="true">
             <intent-filter>
                 <action android:name="android.intent.action.MAIN" />
-
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
+
+        <activity
+            android:name=".activity.GroupStandingsActivity"
+            android:exported="false" />
+
+        <activity
+            android:name=".activity.MainActivity"
+            android:exported="false" />
+
     </application>
 
 </manifest>

--- a/app/src/main/java/com/ventura/emilp/tournamentbrackets/activity/GroupStandingsActivity.java
+++ b/app/src/main/java/com/ventura/emilp/tournamentbrackets/activity/GroupStandingsActivity.java
@@ -1,0 +1,168 @@
+package com.ventura.emilp.tournamentbrackets.activity;
+
+import android.os.Bundle;
+import android.util.Log;
+import android.widget.Toast;
+
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.recyclerview.widget.LinearLayoutManager;
+
+import com.ventura.emilp.tournamentbrackets.adapter.GroupStandingsAdapter;
+import com.ventura.emilp.tournamentbrackets.adapter.GroupStandingsAdapter.GroupData;
+import com.ventura.emilp.tournamentbrackets.databinding.ActivityGroupStandingsBinding;
+import com.ventura.emilp.tournamentbrackets.model.Group;
+import com.ventura.emilp.tournamentbrackets.model.GroupTeamDisplayItem;
+import com.ventura.emilp.tournamentbrackets.model.GroupTeamEntry;
+import com.ventura.emilp.tournamentbrackets.model.GroupsResponse;
+import com.ventura.emilp.tournamentbrackets.model.Team;
+import com.ventura.emilp.tournamentbrackets.model.TeamsResponse;
+import com.ventura.emilp.tournamentbrackets.network.RetrofitClient;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class GroupStandingsActivity extends AppCompatActivity {
+
+    private static final String TAG = "GroupStandingsActivity";
+
+    private ActivityGroupStandingsBinding binding;
+    private GroupStandingsAdapter adapter;
+
+    // Holds results from the two parallel API calls
+    private final AtomicReference<List<Group>> groupsRef = new AtomicReference<>();
+    private final AtomicReference<Map<String, Team>> teamsMapRef = new AtomicReference<>();
+    private final AtomicInteger pendingCalls = new AtomicInteger(2);
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        binding = ActivityGroupStandingsBinding.inflate(getLayoutInflater());
+        setContentView(binding.getRoot());
+
+        if (getSupportActionBar() != null) {
+            getSupportActionBar().setTitle("Group Standings");
+            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+        }
+
+        adapter = new GroupStandingsAdapter();
+        binding.recyclerGroups.setLayoutManager(new LinearLayoutManager(this));
+        binding.recyclerGroups.setAdapter(adapter);
+
+        binding.pBar.setVisibility(android.view.View.VISIBLE);
+        binding.recyclerGroups.setVisibility(android.view.View.GONE);
+
+        fetchGroups();
+        fetchTeams();
+    }
+
+    @Override
+    public boolean onSupportNavigateUp() {
+        finish();
+        return true;
+    }
+
+    private void fetchGroups() {
+        RetrofitClient.getApi().getGroups().enqueue(new Callback<GroupsResponse>() {
+            @Override
+            public void onResponse(Call<GroupsResponse> call, Response<GroupsResponse> response) {
+                if (response.isSuccessful() && response.body() != null) {
+                    groupsRef.set(response.body().getGroups());
+                    Log.d(TAG, "Groups fetched: " + response.body().getGroups().size());
+                } else {
+                    Log.e(TAG, "Groups call failed: " + response.code());
+                }
+                checkBothReady();
+            }
+
+            @Override
+            public void onFailure(Call<GroupsResponse> call, Throwable t) {
+                Log.e(TAG, "Groups fetch error", t);
+                checkBothReady();
+            }
+        });
+    }
+
+    private void fetchTeams() {
+        RetrofitClient.getApi().getTeams().enqueue(new Callback<TeamsResponse>() {
+            @Override
+            public void onResponse(Call<TeamsResponse> call, Response<TeamsResponse> response) {
+                if (response.isSuccessful() && response.body() != null) {
+                    Map<String, Team> map = new HashMap<>();
+                    for (Team t : response.body().getTeams()) {
+                        map.put(t.getId(), t);
+                    }
+                    teamsMapRef.set(map);
+                    Log.d(TAG, "Teams fetched: " + map.size());
+                } else {
+                    Log.e(TAG, "Teams call failed: " + response.code());
+                }
+                checkBothReady();
+            }
+
+            @Override
+            public void onFailure(Call<TeamsResponse> call, Throwable t) {
+                Log.e(TAG, "Teams fetch error", t);
+                checkBothReady();
+            }
+        });
+    }
+
+    /** Called after each completed API call; builds the UI when both are done. */
+    private void checkBothReady() {
+        if (pendingCalls.decrementAndGet() != 0) return;
+
+        List<Group> groups = groupsRef.get();
+        Map<String, Team> teamsMap = teamsMapRef.get();
+
+        if (groups == null || teamsMap == null) {
+            runOnUiThread(() -> {
+                binding.pBar.setVisibility(android.view.View.GONE);
+                Toast.makeText(this, "Failed to load standings", Toast.LENGTH_SHORT).show();
+            });
+            return;
+        }
+
+        // Build display list, sorting groups alphabetically then teams by points desc
+        List<GroupData> displayData = new ArrayList<>();
+        Collections.sort(groups, (a, b) -> a.getName().compareTo(b.getName()));
+
+        for (Group group : groups) {
+            List<GroupTeamEntry> entries = new ArrayList<>(group.getTeams());
+            // Sort by points descending, then goal difference descending
+            Collections.sort(entries, (a, b) -> {
+                int ptsDiff = parseInt(b.getPts()) - parseInt(a.getPts());
+                if (ptsDiff != 0) return ptsDiff;
+                return parseInt(b.getGd()) - parseInt(a.getGd());
+            });
+
+            List<GroupTeamDisplayItem> rowItems = new ArrayList<>();
+            for (GroupTeamEntry entry : entries) {
+                Team team = teamsMap.get(entry.getTeamId());
+                if (team != null) {
+                    rowItems.add(new GroupTeamDisplayItem(team, entry));
+                }
+            }
+            displayData.add(new GroupData(group.getName(), rowItems));
+        }
+
+        runOnUiThread(() -> {
+            binding.pBar.setVisibility(android.view.View.GONE);
+            binding.recyclerGroups.setVisibility(android.view.View.VISIBLE);
+            adapter.setData(displayData);
+        });
+    }
+
+    private static int parseInt(String s) {
+        if (s == null) return 0;
+        try { return Integer.parseInt(s); } catch (NumberFormatException e) { return 0; }
+    }
+}

--- a/app/src/main/java/com/ventura/emilp/tournamentbrackets/activity/HomeActivity.java
+++ b/app/src/main/java/com/ventura/emilp/tournamentbrackets/activity/HomeActivity.java
@@ -1,0 +1,26 @@
+package com.ventura.emilp.tournamentbrackets.activity;
+
+import android.content.Intent;
+import android.os.Bundle;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.ventura.emilp.tournamentbrackets.databinding.ActivityHomeBinding;
+
+public class HomeActivity extends AppCompatActivity {
+
+    private ActivityHomeBinding binding;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        binding = ActivityHomeBinding.inflate(getLayoutInflater());
+        setContentView(binding.getRoot());
+
+        binding.cardGroupStandings.setOnClickListener(v ->
+                startActivity(new Intent(this, GroupStandingsActivity.class)));
+
+        binding.cardBrackets.setOnClickListener(v ->
+                startActivity(new Intent(this, MainActivity.class)));
+    }
+}

--- a/app/src/main/java/com/ventura/emilp/tournamentbrackets/activity/MainActivity.java
+++ b/app/src/main/java/com/ventura/emilp/tournamentbrackets/activity/MainActivity.java
@@ -17,13 +17,19 @@ import com.ventura.bracketslib.model.ColomnData;
 import com.ventura.bracketslib.model.CompetitorData;
 import com.ventura.bracketslib.model.MatchData;
 import com.ventura.emilp.tournamentbrackets.databinding.ActivityMainBinding;
+import com.ventura.emilp.tournamentbrackets.model.Team;
+import com.ventura.emilp.tournamentbrackets.model.TeamsResponse;
 import com.ventura.emilp.tournamentbrackets.model.WorldCupGame;
 import com.ventura.emilp.tournamentbrackets.model.WorldCupResponse;
 import com.ventura.emilp.tournamentbrackets.network.RetrofitClient;
 
 import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
 
 import retrofit2.Call;
 import retrofit2.Callback;
@@ -34,95 +40,165 @@ public class MainActivity extends AppCompatActivity {
     private ActivityMainBinding binding;
     private static final String TAG = "MainActivity";
 
+    private final AtomicReference<List<WorldCupGame>> gamesRef = new AtomicReference<>();
+    private final AtomicReference<Map<String, Team>> teamsMapRef = new AtomicReference<>();
+    private final AtomicInteger pendingCalls = new AtomicInteger(2);
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setScreenSize();
         binding = ActivityMainBinding.inflate(getLayoutInflater());
         setContentView(binding.getRoot());
+
+        if (getSupportActionBar() != null) {
+            getSupportActionBar().setTitle("Knockout Brackets");
+            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+        }
         binding.pBar.setVisibility(android.view.View.VISIBLE);
-        binding.bracketView.postDelayed(new Runnable() {
-            @Override
-            public void run() {
-                fetchGames();
-            }
-        }, 1000);
+        binding.bracketView.setVisibility(android.view.View.GONE);
+        fetchGames();
+        fetchTeams();
     }
 
     private void fetchGames() {
         RetrofitClient.getApi().getGames().enqueue(new Callback<WorldCupResponse>() {
             @Override
             public void onResponse(Call<WorldCupResponse> call, Response<WorldCupResponse> response) {
-                binding.pBar.setVisibility(android.view.View.GONE);
                 if (response.isSuccessful() && response.body() != null) {
-                    List<WorldCupGame> games = response.body().getGames();
-                    Log.d(TAG, "Fetched " + (games != null ? games.size() : 0) + " games");
-                    processGames(games);
+                    gamesRef.set(response.body().getGames());
+                    Log.d(TAG, "Fetched " + response.body().getGames().size() + " games");
                 } else {
                     Log.e(TAG, "Failed to fetch games: " + response.code());
-                    Toast.makeText(MainActivity.this, "Failed to fetch games", Toast.LENGTH_SHORT).show();
                 }
+                checkBothReady();
             }
 
             @Override
             public void onFailure(Call<WorldCupResponse> call, Throwable t) {
-                binding.pBar.setVisibility(android.view.View.GONE);
                 Log.e(TAG, "Error fetching games", t);
-                Toast.makeText(MainActivity.this, "Error: " + t.getMessage(), Toast.LENGTH_SHORT).show();
+                checkBothReady();
             }
         });
     }
 
-    private void processGames(List<WorldCupGame> games) {
-        if (games == null) return;
-        Toast.makeText(this, "Processing " + games.size() + " games", Toast.LENGTH_SHORT).show();
-        List<ColomnData> colomnDataList = new ArrayList<>();
+    private void fetchTeams() {
+        RetrofitClient.getApi().getTeams().enqueue(new Callback<TeamsResponse>() {
+            @Override
+            public void onResponse(Call<TeamsResponse> call, Response<TeamsResponse> response) {
+                if (response.isSuccessful() && response.body() != null) {
+                    Map<String, Team> map = new HashMap<>();
+                    for (Team t : response.body().getTeams()) {
+                        map.put(t.getId(), t);
+                    }
+                    teamsMapRef.set(map);
+                    Log.d(TAG, "Fetched " + map.size() + " teams");
+                } else {
+                    Log.e(TAG, "Failed to fetch teams: " + response.code());
+                }
+                checkBothReady();
+            }
 
-        colomnDataList.add(getColomnDataForType(games, "r32", Arrays.asList("74", "77", "73", "75", "83", "84", "81", "82", "76", "78", "79", "80", "86", "88", "85", "87")));
-        colomnDataList.add(getColomnDataForType(games, "r16", Arrays.asList("89", "90", "93", "94", "91", "92", "95", "96")));
-        colomnDataList.add(getColomnDataForType(games, "qf", Arrays.asList("97", "98", "99", "100")));
-        colomnDataList.add(getColomnDataForType(games, "sf", Arrays.asList("101", "102")));
-        colomnDataList.add(getColomnDataForType(games, "final", Arrays.asList("104")));
+            @Override
+            public void onFailure(Call<TeamsResponse> call, Throwable t) {
+                Log.e(TAG, "Error fetching teams", t);
+                checkBothReady();
+            }
+        });
+    }
 
-        Log.d(TAG, "Setting brackets data with " + colomnDataList.size() + " columns");
-        for (int j = 0; j < colomnDataList.size(); j++) {
-            Log.d(TAG, "Column " + j + " matches: " + colomnDataList.get(j).getMatches().size());
-        }
+    private void checkBothReady() {
+        if (pendingCalls.decrementAndGet() != 0) return;
+
+        List<WorldCupGame> games = gamesRef.get();
+        Map<String, Team> teamsMap = teamsMapRef.get();
 
         runOnUiThread(new Runnable() {
             @Override
             public void run() {
-                binding.bracketView.setBracketsData(colomnDataList);
+                binding.pBar.setVisibility(android.view.View.GONE);
+                if (games == null) {
+                    Toast.makeText(MainActivity.this, "Failed to load games", Toast.LENGTH_SHORT).show();
+                    return;
+                }
+                binding.bracketView.setVisibility(android.view.View.VISIBLE);
+                processGames(games, teamsMap);
             }
         });
     }
 
-    private ColomnData getColomnDataForType(List<WorldCupGame> games, String type, List<String> orderedIds) {
+    private void processGames(List<WorldCupGame> games, Map<String, Team> teamsMap) {
+        List<ColomnData> colomnDataList = new ArrayList<>();
+
+        colomnDataList.add(getColomnDataForType(games, teamsMap, "r32", Arrays.asList("74", "77", "73", "75", "83", "84", "81", "82", "76", "78", "79", "80", "86", "88", "85", "87")));
+        colomnDataList.add(getColomnDataForType(games, teamsMap, "r16", Arrays.asList("89", "90", "93", "94", "91", "92", "95", "96")));
+        colomnDataList.add(getColomnDataForType(games, teamsMap, "qf", Arrays.asList("97", "98", "99", "100")));
+        colomnDataList.add(getColomnDataForType(games, teamsMap, "sf", Arrays.asList("101", "102")));
+        colomnDataList.add(getColomnDataForType(games, teamsMap, "final", Arrays.asList("104")));
+
+        Log.d(TAG, "Setting brackets data with " + colomnDataList.size() + " columns");
+        binding.bracketView.setBracketsData(colomnDataList);
+    }
+
+    private ColomnData getColomnDataForType(List<WorldCupGame> games, Map<String, Team> teamsMap, String type, List<String> orderedIds) {
         List<MatchData> matches = new ArrayList<>();
 
         for (String id : orderedIds) {
             for (WorldCupGame game : games) {
                 if (id.equals(game.getId())) {
-                    CompetitorData home = new CompetitorData(game.getHomeTeam(), game.getHomeScore());
-                    CompetitorData away = new CompetitorData(game.getAwayTeam(), game.getAwayScore());
-                    matches.add(new MatchData(home, away));
+                    matches.add(createMatchData(game, teamsMap));
                     break;
                 }
             }
         }
 
-        // Fallback if some games are missing or we want to show all games of that type
+        // Fallback if some games are missing
         if (matches.isEmpty()) {
             for (WorldCupGame game : games) {
                 if (type.equals(game.getType())) {
-                    CompetitorData home = new CompetitorData(game.getHomeTeam(), game.getHomeScore());
-                    CompetitorData away = new CompetitorData(game.getAwayTeam(), game.getAwayScore());
-                    matches.add(new MatchData(home, away));
+                    matches.add(createMatchData(game, teamsMap));
                 }
             }
         }
 
         return new ColomnData(matches);
+    }
+
+    private MatchData createMatchData(WorldCupGame game, Map<String, Team> teamsMap) {
+        CompetitorData home = new CompetitorData(game.getHomeTeam(), game.getHomeScore());
+        CompetitorData away = new CompetitorData(game.getAwayTeam(), game.getAwayScore());
+
+        if (teamsMap != null) {
+            Team homeTeam = teamsMap.get(game.getHomeTeamId());
+            Team awayTeam = teamsMap.get(game.getAwayTeamId());
+            if (homeTeam != null) {
+                home.setImageUrl(homeTeam.getFlagUrl());
+            }
+            if (awayTeam != null) {
+                away.setImageUrl(awayTeam.getFlagUrl());
+            }
+        }
+
+        MatchData match = new MatchData(home, away);
+        match.setMatchName(getMatchName(game));
+        return match;
+    }
+
+    private String getMatchName(WorldCupGame game) {
+        switch (game.getType()) {
+            case "r32": return "R32 - Match " + game.getId();
+            case "r16": return "R16 - Match " + game.getId();
+            case "qf": return "QF - Match " + game.getId();
+            case "sf": return "SF - Match " + game.getId();
+            case "final": return "Final";
+            default: return "Match " + game.getId();
+        }
+    }
+
+    @Override
+    public boolean onSupportNavigateUp() {
+        finish();
+        return true;
     }
 
     @Override
@@ -139,7 +215,6 @@ public class MainActivity extends AppCompatActivity {
             int height = windowMetrics.getBounds().height() - insets.top - insets.bottom;
             BracketsConfig.getInstance().setScreenHeight(height);
         } else {
-            // Support for devices below API 30 (minimum SDK is 24)
             DisplayMetrics displayMetrics = new DisplayMetrics();
             getWindowManager().getDefaultDisplay().getMetrics(displayMetrics);
             int height = displayMetrics.heightPixels;

--- a/app/src/main/java/com/ventura/emilp/tournamentbrackets/activity/MainActivity.java
+++ b/app/src/main/java/com/ventura/emilp/tournamentbrackets/activity/MainActivity.java
@@ -7,6 +7,9 @@ import android.util.DisplayMetrics;
 import android.view.WindowInsets;
 import android.view.WindowMetrics;
 
+import android.util.Log;
+import android.widget.Toast;
+
 import androidx.appcompat.app.AppCompatActivity;
 
 import com.ventura.bracketslib.application.BracketsConfig;
@@ -14,37 +17,112 @@ import com.ventura.bracketslib.model.ColomnData;
 import com.ventura.bracketslib.model.CompetitorData;
 import com.ventura.bracketslib.model.MatchData;
 import com.ventura.emilp.tournamentbrackets.databinding.ActivityMainBinding;
+import com.ventura.emilp.tournamentbrackets.model.WorldCupGame;
+import com.ventura.emilp.tournamentbrackets.model.WorldCupResponse;
+import com.ventura.emilp.tournamentbrackets.network.RetrofitClient;
 
+import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.List;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
 
 public class MainActivity extends AppCompatActivity {
 
     private ActivityMainBinding binding;
+    private static final String TAG = "MainActivity";
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        setScreenSize();
         binding = ActivityMainBinding.inflate(getLayoutInflater());
         setContentView(binding.getRoot());
-        setupBrackets();
+        binding.pBar.setVisibility(android.view.View.VISIBLE);
+        binding.bracketView.postDelayed(new Runnable() {
+            @Override
+            public void run() {
+                fetchGames();
+            }
+        }, 1000);
     }
 
-    private void setupBrackets() {
-        CompetitorData brazilSemiFinal = new CompetitorData("Brazil", "3");
-        CompetitorData englandSemiFinal = new CompetitorData("England", "1");
-        CompetitorData argentinaSemiFinal = new CompetitorData("Argentina", "3");
-        CompetitorData russiaSemiFinal = new CompetitorData("Russia", "2");
-        CompetitorData brazilFinal = new CompetitorData("Brazil", "4");
-        CompetitorData argentinaFinal = new CompetitorData("Argentina", "2");
+    private void fetchGames() {
+        RetrofitClient.getApi().getGames().enqueue(new Callback<WorldCupResponse>() {
+            @Override
+            public void onResponse(Call<WorldCupResponse> call, Response<WorldCupResponse> response) {
+                binding.pBar.setVisibility(android.view.View.GONE);
+                if (response.isSuccessful() && response.body() != null) {
+                    List<WorldCupGame> games = response.body().getGames();
+                    Log.d(TAG, "Fetched " + (games != null ? games.size() : 0) + " games");
+                    processGames(games);
+                } else {
+                    Log.e(TAG, "Failed to fetch games: " + response.code());
+                    Toast.makeText(MainActivity.this, "Failed to fetch games", Toast.LENGTH_SHORT).show();
+                }
+            }
 
-        MatchData match1SemiFinal = new MatchData(brazilSemiFinal, englandSemiFinal);
-        MatchData match2SemiFinal = new MatchData(argentinaSemiFinal, russiaSemiFinal);
-        MatchData match3Final = new MatchData(brazilFinal, argentinaFinal);
+            @Override
+            public void onFailure(Call<WorldCupResponse> call, Throwable t) {
+                binding.pBar.setVisibility(android.view.View.GONE);
+                Log.e(TAG, "Error fetching games", t);
+                Toast.makeText(MainActivity.this, "Error: " + t.getMessage(), Toast.LENGTH_SHORT).show();
+            }
+        });
+    }
 
-        ColomnData semiFinalColomn = new ColomnData(Arrays.asList(match1SemiFinal, match2SemiFinal));
-        ColomnData finalColomn = new ColomnData(Arrays.asList(match3Final));
+    private void processGames(List<WorldCupGame> games) {
+        if (games == null) return;
+        Toast.makeText(this, "Processing " + games.size() + " games", Toast.LENGTH_SHORT).show();
+        List<ColomnData> colomnDataList = new ArrayList<>();
 
-        binding.bracketView.setBracketsData(Arrays.asList(semiFinalColomn, finalColomn));
+        colomnDataList.add(getColomnDataForType(games, "r32", Arrays.asList("74", "77", "73", "75", "83", "84", "81", "82", "76", "78", "79", "80", "86", "88", "85", "87")));
+        colomnDataList.add(getColomnDataForType(games, "r16", Arrays.asList("89", "90", "93", "94", "91", "92", "95", "96")));
+        colomnDataList.add(getColomnDataForType(games, "qf", Arrays.asList("97", "98", "99", "100")));
+        colomnDataList.add(getColomnDataForType(games, "sf", Arrays.asList("101", "102")));
+        colomnDataList.add(getColomnDataForType(games, "final", Arrays.asList("104")));
+
+        Log.d(TAG, "Setting brackets data with " + colomnDataList.size() + " columns");
+        for (int j = 0; j < colomnDataList.size(); j++) {
+            Log.d(TAG, "Column " + j + " matches: " + colomnDataList.get(j).getMatches().size());
+        }
+
+        runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+                binding.bracketView.setBracketsData(colomnDataList);
+            }
+        });
+    }
+
+    private ColomnData getColomnDataForType(List<WorldCupGame> games, String type, List<String> orderedIds) {
+        List<MatchData> matches = new ArrayList<>();
+
+        for (String id : orderedIds) {
+            for (WorldCupGame game : games) {
+                if (id.equals(game.getId())) {
+                    CompetitorData home = new CompetitorData(game.getHomeTeam(), game.getHomeScore());
+                    CompetitorData away = new CompetitorData(game.getAwayTeam(), game.getAwayScore());
+                    matches.add(new MatchData(home, away));
+                    break;
+                }
+            }
+        }
+
+        // Fallback if some games are missing or we want to show all games of that type
+        if (matches.isEmpty()) {
+            for (WorldCupGame game : games) {
+                if (type.equals(game.getType())) {
+                    CompetitorData home = new CompetitorData(game.getHomeTeam(), game.getHomeScore());
+                    CompetitorData away = new CompetitorData(game.getAwayTeam(), game.getAwayScore());
+                    matches.add(new MatchData(home, away));
+                }
+            }
+        }
+
+        return new ColomnData(matches);
     }
 
     @Override

--- a/app/src/main/java/com/ventura/emilp/tournamentbrackets/adapter/GroupStandingsAdapter.java
+++ b/app/src/main/java/com/ventura/emilp/tournamentbrackets/adapter/GroupStandingsAdapter.java
@@ -1,0 +1,109 @@
+package com.ventura.emilp.tournamentbrackets.adapter;
+
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.bumptech.glide.Glide;
+import com.ventura.emilp.tournamentbrackets.R;
+import com.ventura.emilp.tournamentbrackets.model.GroupTeamDisplayItem;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class GroupStandingsAdapter extends RecyclerView.Adapter<GroupStandingsAdapter.GroupViewHolder> {
+
+    private final List<GroupData> groupList = new ArrayList<>();
+
+    public void setData(List<GroupData> data) {
+        groupList.clear();
+        groupList.addAll(data);
+        notifyDataSetChanged();
+    }
+
+    @NonNull
+    @Override
+    public GroupViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(parent.getContext())
+                .inflate(R.layout.item_group_table, parent, false);
+        return new GroupViewHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull GroupViewHolder holder, int position) {
+        holder.bind(groupList.get(position));
+    }
+
+    @Override
+    public int getItemCount() {
+        return groupList.size();
+    }
+
+    static class GroupViewHolder extends RecyclerView.ViewHolder {
+        private final TextView tvGroupName;
+        private final LinearLayout llTeamsContainer;
+
+        GroupViewHolder(@NonNull View itemView) {
+            super(itemView);
+            tvGroupName = itemView.findViewById(R.id.tv_group_name);
+            llTeamsContainer = itemView.findViewById(R.id.ll_teams_container);
+        }
+
+        void bind(GroupData groupData) {
+            tvGroupName.setText("Group " + groupData.groupName);
+            llTeamsContainer.removeAllViews();
+
+            Context context = itemView.getContext();
+            LayoutInflater inflater = LayoutInflater.from(context);
+
+            for (GroupTeamDisplayItem item : groupData.teams) {
+                View row = inflater.inflate(R.layout.item_group_team_row, llTeamsContainer, false);
+
+                ImageView ivFlag = row.findViewById(R.id.iv_flag);
+                TextView tvName = row.findViewById(R.id.tv_name);
+                TextView tvMp = row.findViewById(R.id.tv_mp);
+                TextView tvW = row.findViewById(R.id.tv_w);
+                TextView tvD = row.findViewById(R.id.tv_d);
+                TextView tvL = row.findViewById(R.id.tv_l);
+                TextView tvGf = row.findViewById(R.id.tv_gf);
+                TextView tvGa = row.findViewById(R.id.tv_ga);
+                TextView tvGd = row.findViewById(R.id.tv_gd);
+                TextView tvPts = row.findViewById(R.id.tv_pts);
+
+                Glide.with(context)
+                        .load(item.getFlagUrl())
+                        .into(ivFlag);
+
+                tvName.setText(item.getName());
+                tvMp.setText(item.getMp());
+                tvW.setText(item.getW());
+                tvD.setText(item.getD());
+                tvL.setText(item.getL());
+                tvGf.setText(item.getGf());
+                tvGa.setText(item.getGa());
+                tvGd.setText(item.getGd());
+                tvPts.setText(item.getPts());
+
+                llTeamsContainer.addView(row);
+            }
+        }
+    }
+
+    /** Simple data holder for one group's display data. */
+    public static class GroupData {
+        public final String groupName;
+        public final List<GroupTeamDisplayItem> teams;
+
+        public GroupData(String groupName, List<GroupTeamDisplayItem> teams) {
+            this.groupName = groupName;
+            this.teams = teams;
+        }
+    }
+}

--- a/app/src/main/java/com/ventura/emilp/tournamentbrackets/model/Group.java
+++ b/app/src/main/java/com/ventura/emilp/tournamentbrackets/model/Group.java
@@ -1,0 +1,20 @@
+package com.ventura.emilp.tournamentbrackets.model;
+
+import com.google.gson.annotations.SerializedName;
+import java.util.List;
+
+public class Group {
+    @SerializedName("name")
+    private String name;
+
+    @SerializedName("teams")
+    private List<GroupTeamEntry> teams;
+
+    public String getName() {
+        return name;
+    }
+
+    public List<GroupTeamEntry> getTeams() {
+        return teams;
+    }
+}

--- a/app/src/main/java/com/ventura/emilp/tournamentbrackets/model/GroupTeamDisplayItem.java
+++ b/app/src/main/java/com/ventura/emilp/tournamentbrackets/model/GroupTeamDisplayItem.java
@@ -1,0 +1,39 @@
+package com.ventura.emilp.tournamentbrackets.model;
+
+/** Merged display model combining GroupTeamEntry stats with Team name/flag. */
+public class GroupTeamDisplayItem {
+    private final String name;
+    private final String flagUrl;
+    private final String mp;
+    private final String w;
+    private final String d;
+    private final String l;
+    private final String gf;
+    private final String ga;
+    private final String gd;
+    private final String pts;
+
+    public GroupTeamDisplayItem(Team team, GroupTeamEntry entry) {
+        this.name = team.getNameEn();
+        this.flagUrl = team.getFlagUrl();
+        this.mp = entry.getMp();
+        this.w = entry.getW();
+        this.d = entry.getD();
+        this.l = entry.getL();
+        this.gf = entry.getGf();
+        this.ga = entry.getGa();
+        this.gd = entry.getGd();
+        this.pts = entry.getPts();
+    }
+
+    public String getName() { return name; }
+    public String getFlagUrl() { return flagUrl; }
+    public String getMp() { return mp; }
+    public String getW() { return w; }
+    public String getD() { return d; }
+    public String getL() { return l; }
+    public String getGf() { return gf; }
+    public String getGa() { return ga; }
+    public String getGd() { return gd; }
+    public String getPts() { return pts; }
+}

--- a/app/src/main/java/com/ventura/emilp/tournamentbrackets/model/GroupTeamEntry.java
+++ b/app/src/main/java/com/ventura/emilp/tournamentbrackets/model/GroupTeamEntry.java
@@ -1,0 +1,42 @@
+package com.ventura.emilp.tournamentbrackets.model;
+
+import com.google.gson.annotations.SerializedName;
+
+public class GroupTeamEntry {
+    @SerializedName("team_id")
+    private String teamId;
+
+    @SerializedName("mp")
+    private String mp;
+
+    @SerializedName("w")
+    private String w;
+
+    @SerializedName("d")
+    private String d;
+
+    @SerializedName("l")
+    private String l;
+
+    @SerializedName("gf")
+    private String gf;
+
+    @SerializedName("ga")
+    private String ga;
+
+    @SerializedName("gd")
+    private String gd;
+
+    @SerializedName("pts")
+    private String pts;
+
+    public String getTeamId() { return teamId; }
+    public String getMp() { return mp; }
+    public String getW() { return w; }
+    public String getD() { return d; }
+    public String getL() { return l; }
+    public String getGf() { return gf; }
+    public String getGa() { return ga; }
+    public String getGd() { return gd; }
+    public String getPts() { return pts; }
+}

--- a/app/src/main/java/com/ventura/emilp/tournamentbrackets/model/GroupsResponse.java
+++ b/app/src/main/java/com/ventura/emilp/tournamentbrackets/model/GroupsResponse.java
@@ -1,0 +1,13 @@
+package com.ventura.emilp.tournamentbrackets.model;
+
+import com.google.gson.annotations.SerializedName;
+import java.util.List;
+
+public class GroupsResponse {
+    @SerializedName("groups")
+    private List<Group> groups;
+
+    public List<Group> getGroups() {
+        return groups;
+    }
+}

--- a/app/src/main/java/com/ventura/emilp/tournamentbrackets/model/Team.java
+++ b/app/src/main/java/com/ventura/emilp/tournamentbrackets/model/Team.java
@@ -1,0 +1,22 @@
+package com.ventura.emilp.tournamentbrackets.model;
+
+import com.google.gson.annotations.SerializedName;
+
+public class Team {
+    @SerializedName("id")
+    private String id;
+
+    @SerializedName("name_en")
+    private String nameEn;
+
+    @SerializedName("flag")
+    private String flagUrl;
+
+    @SerializedName("fifa_code")
+    private String fifaCode;
+
+    public String getId() { return id; }
+    public String getNameEn() { return nameEn; }
+    public String getFlagUrl() { return flagUrl; }
+    public String getFifaCode() { return fifaCode; }
+}

--- a/app/src/main/java/com/ventura/emilp/tournamentbrackets/model/TeamsResponse.java
+++ b/app/src/main/java/com/ventura/emilp/tournamentbrackets/model/TeamsResponse.java
@@ -1,0 +1,13 @@
+package com.ventura.emilp.tournamentbrackets.model;
+
+import com.google.gson.annotations.SerializedName;
+import java.util.List;
+
+public class TeamsResponse {
+    @SerializedName("teams")
+    private List<Team> teams;
+
+    public List<Team> getTeams() {
+        return teams;
+    }
+}

--- a/app/src/main/java/com/ventura/emilp/tournamentbrackets/model/WorldCupGame.java
+++ b/app/src/main/java/com/ventura/emilp/tournamentbrackets/model/WorldCupGame.java
@@ -30,6 +30,12 @@ public class WorldCupGame {
     @SerializedName("finished")
     private String finished;
 
+    @SerializedName("home_team_id")
+    private String homeTeamId;
+
+    @SerializedName("away_team_id")
+    private String awayTeamId;
+
     public String getId() {
         return id;
     }
@@ -56,5 +62,13 @@ public class WorldCupGame {
 
     public String getType() {
         return type;
+    }
+
+    public String getHomeTeamId() {
+        return homeTeamId;
+    }
+
+    public String getAwayTeamId() {
+        return awayTeamId;
     }
 }

--- a/app/src/main/java/com/ventura/emilp/tournamentbrackets/model/WorldCupGame.java
+++ b/app/src/main/java/com/ventura/emilp/tournamentbrackets/model/WorldCupGame.java
@@ -1,0 +1,60 @@
+package com.ventura.emilp.tournamentbrackets.model;
+
+import com.google.gson.annotations.SerializedName;
+
+public class WorldCupGame {
+    @SerializedName("id")
+    private String id;
+
+    @SerializedName("home_team_name_en")
+    private String homeTeam;
+
+    @SerializedName("away_team_name_en")
+    private String awayTeam;
+
+    @SerializedName("home_score")
+    private String homeScore;
+
+    @SerializedName("away_score")
+    private String awayScore;
+
+    @SerializedName("type")
+    private String type;
+
+    @SerializedName("home_team_label")
+    private String homeTeamLabel;
+
+    @SerializedName("away_team_label")
+    private String awayTeamLabel;
+
+    @SerializedName("finished")
+    private String finished;
+
+    public String getId() {
+        return id;
+    }
+
+    public String getHomeTeam() {
+        return homeTeam != null ? homeTeam : homeTeamLabel;
+    }
+
+    public String getAwayTeam() {
+        return awayTeam != null ? awayTeam : awayTeamLabel;
+    }
+
+    public boolean isFinished() {
+        return "TRUE".equalsIgnoreCase(finished);
+    }
+
+    public String getHomeScore() {
+        return isFinished() ? homeScore : null;
+    }
+
+    public String getAwayScore() {
+        return isFinished() ? awayScore : null;
+    }
+
+    public String getType() {
+        return type;
+    }
+}

--- a/app/src/main/java/com/ventura/emilp/tournamentbrackets/model/WorldCupResponse.java
+++ b/app/src/main/java/com/ventura/emilp/tournamentbrackets/model/WorldCupResponse.java
@@ -1,0 +1,13 @@
+package com.ventura.emilp.tournamentbrackets.model;
+
+import com.google.gson.annotations.SerializedName;
+import java.util.List;
+
+public class WorldCupResponse {
+    @SerializedName("games")
+    private List<WorldCupGame> games;
+
+    public List<WorldCupGame> getGames() {
+        return games;
+    }
+}

--- a/app/src/main/java/com/ventura/emilp/tournamentbrackets/network/RetrofitClient.java
+++ b/app/src/main/java/com/ventura/emilp/tournamentbrackets/network/RetrofitClient.java
@@ -1,0 +1,26 @@
+package com.ventura.emilp.tournamentbrackets.network;
+
+import okhttp3.OkHttpClient;
+import okhttp3.logging.HttpLoggingInterceptor;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+public class RetrofitClient {
+    private static final String BASE_URL = "https://worldcup26.ir/";
+    private static Retrofit retrofit = null;
+
+    public static WorldCupApi getApi() {
+        if (retrofit == null) {
+            HttpLoggingInterceptor interceptor = new HttpLoggingInterceptor();
+            interceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
+            OkHttpClient client = new OkHttpClient.Builder().addInterceptor(interceptor).build();
+
+            retrofit = new Retrofit.Builder()
+                    .baseUrl(BASE_URL)
+                    .addConverterFactory(GsonConverterFactory.create())
+                    .client(client)
+                    .build();
+        }
+        return retrofit.create(WorldCupApi.class);
+    }
+}

--- a/app/src/main/java/com/ventura/emilp/tournamentbrackets/network/WorldCupApi.java
+++ b/app/src/main/java/com/ventura/emilp/tournamentbrackets/network/WorldCupApi.java
@@ -1,5 +1,7 @@
 package com.ventura.emilp.tournamentbrackets.network;
 
+import com.ventura.emilp.tournamentbrackets.model.GroupsResponse;
+import com.ventura.emilp.tournamentbrackets.model.TeamsResponse;
 import com.ventura.emilp.tournamentbrackets.model.WorldCupResponse;
 
 import retrofit2.Call;
@@ -8,4 +10,10 @@ import retrofit2.http.GET;
 public interface WorldCupApi {
     @GET("get/games")
     Call<WorldCupResponse> getGames();
+
+    @GET("get/groups")
+    Call<GroupsResponse> getGroups();
+
+    @GET("get/teams")
+    Call<TeamsResponse> getTeams();
 }

--- a/app/src/main/java/com/ventura/emilp/tournamentbrackets/network/WorldCupApi.java
+++ b/app/src/main/java/com/ventura/emilp/tournamentbrackets/network/WorldCupApi.java
@@ -1,0 +1,11 @@
+package com.ventura.emilp.tournamentbrackets.network;
+
+import com.ventura.emilp.tournamentbrackets.model.WorldCupResponse;
+
+import retrofit2.Call;
+import retrofit2.http.GET;
+
+public interface WorldCupApi {
+    @GET("get/games")
+    Call<WorldCupResponse> getGames();
+}

--- a/app/src/main/res/layout/activity_group_standings.xml
+++ b/app/src/main/res/layout/activity_group_standings.xml
@@ -1,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:background="#1c222e">
+
+    <ProgressBar
+        android:id="@+id/pBar"
+        android:layout_centerInParent="true"
+        android:visibility="gone"
+        android:layout_width="50dp"
+        android:layout_height="50dp" />
+
+    <androidx.recyclerview.widget.RecyclerView
+        android:id="@+id/recycler_groups"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:padding="12dp"
+        android:clipToPadding="false"
+        android:visibility="gone" />
+
+</RelativeLayout>

--- a/app/src/main/res/layout/activity_home.xml
+++ b/app/src/main/res/layout/activity_home.xml
@@ -1,0 +1,100 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical"
+    android:gravity="center"
+    android:padding="32dp"
+    android:background="#1c222e">
+
+    <TextView
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="@string/app_name"
+        android:textColor="#ffffff"
+        android:textSize="26sp"
+        android:textStyle="bold"
+        android:layout_marginBottom="8dp" />
+
+    <TextView
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="@string/home_subtitle"
+        android:textColor="#8a93a8"
+        android:textSize="14sp"
+        android:layout_marginBottom="48dp" />
+
+    <!-- Group Standings Card -->
+    <androidx.cardview.widget.CardView
+        android:id="@+id/card_group_standings"
+        android:layout_width="match_parent"
+        android:layout_height="100dp"
+        android:layout_marginBottom="20dp"
+        app:cardBackgroundColor="#262e40"
+        app:cardCornerRadius="12dp"
+        app:cardElevation="4dp"
+        xmlns:app="http://schemas.android.com/apk/res-auto">
+
+        <LinearLayout
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:orientation="horizontal"
+            android:gravity="center_vertical"
+            android:padding="20dp">
+
+            <TextView
+                android:layout_width="0dp"
+                android:layout_height="wrap_content"
+                android:layout_weight="1"
+                android:text="@string/home_group_standings"
+                android:textColor="#ffffff"
+                android:textSize="18sp"
+                android:textStyle="bold" />
+
+            <TextView
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:text="›"
+                android:textColor="#8a93a8"
+                android:textSize="24sp" />
+
+        </LinearLayout>
+    </androidx.cardview.widget.CardView>
+
+    <!-- Brackets Card -->
+    <androidx.cardview.widget.CardView
+        android:id="@+id/card_brackets"
+        android:layout_width="match_parent"
+        android:layout_height="100dp"
+        app:cardBackgroundColor="#262e40"
+        app:cardCornerRadius="12dp"
+        app:cardElevation="4dp"
+        xmlns:app="http://schemas.android.com/apk/res-auto">
+
+        <LinearLayout
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:orientation="horizontal"
+            android:gravity="center_vertical"
+            android:padding="20dp">
+
+            <TextView
+                android:layout_width="0dp"
+                android:layout_height="wrap_content"
+                android:layout_weight="1"
+                android:text="@string/home_brackets"
+                android:textColor="#ffffff"
+                android:textSize="18sp"
+                android:textStyle="bold" />
+
+            <TextView
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:text="›"
+                android:textColor="#8a93a8"
+                android:textSize="24sp" />
+
+        </LinearLayout>
+    </androidx.cardview.widget.CardView>
+
+</LinearLayout>

--- a/app/src/main/res/layout/activity_main.xml
+++ b/app/src/main/res/layout/activity_main.xml
@@ -2,7 +2,8 @@
 <RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto"
     android:layout_width="match_parent"
-    android:layout_height="match_parent">
+    android:layout_height="match_parent"
+    android:background="#1c222e">
 
     <ProgressBar
         android:id="@+id/pBar"

--- a/app/src/main/res/layout/item_group_table.xml
+++ b/app/src/main/res/layout/item_group_table.xml
@@ -1,0 +1,128 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.cardview.widget.CardView xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:layout_marginBottom="16dp"
+    app:cardBackgroundColor="#262e40"
+    app:cardCornerRadius="10dp"
+    app:cardElevation="2dp">
+
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:orientation="vertical"
+        android:padding="12dp">
+
+        <!-- Group title -->
+        <TextView
+            android:id="@+id/tv_group_name"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:textColor="#ffffff"
+            android:textSize="15sp"
+            android:textStyle="bold"
+            android:layout_marginBottom="10dp" />
+
+        <!-- Column headers -->
+        <LinearLayout
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:orientation="horizontal"
+            android:layout_marginBottom="4dp">
+
+            <ImageView
+                android:layout_width="24dp"
+                android:layout_height="24dp"
+                android:visibility="invisible" />
+
+            <TextView
+                android:layout_width="0dp"
+                android:layout_height="wrap_content"
+                android:layout_weight="1"
+                android:text=""
+                android:layout_marginStart="8dp" />
+
+            <TextView
+                android:layout_width="28dp"
+                android:layout_height="wrap_content"
+                android:text="@string/col_mp"
+                android:textColor="#8a93a8"
+                android:textSize="11sp"
+                android:gravity="center" />
+
+            <TextView
+                android:layout_width="28dp"
+                android:layout_height="wrap_content"
+                android:text="@string/col_w"
+                android:textColor="#8a93a8"
+                android:textSize="11sp"
+                android:gravity="center" />
+
+            <TextView
+                android:layout_width="28dp"
+                android:layout_height="wrap_content"
+                android:text="@string/col_d"
+                android:textColor="#8a93a8"
+                android:textSize="11sp"
+                android:gravity="center" />
+
+            <TextView
+                android:layout_width="28dp"
+                android:layout_height="wrap_content"
+                android:text="@string/col_l"
+                android:textColor="#8a93a8"
+                android:textSize="11sp"
+                android:gravity="center" />
+
+            <TextView
+                android:layout_width="28dp"
+                android:layout_height="wrap_content"
+                android:text="@string/col_gf"
+                android:textColor="#8a93a8"
+                android:textSize="11sp"
+                android:gravity="center" />
+
+            <TextView
+                android:layout_width="28dp"
+                android:layout_height="wrap_content"
+                android:text="@string/col_ga"
+                android:textColor="#8a93a8"
+                android:textSize="11sp"
+                android:gravity="center" />
+
+            <TextView
+                android:layout_width="28dp"
+                android:layout_height="wrap_content"
+                android:text="@string/col_gd"
+                android:textColor="#8a93a8"
+                android:textSize="11sp"
+                android:gravity="center" />
+
+            <TextView
+                android:layout_width="32dp"
+                android:layout_height="wrap_content"
+                android:text="@string/col_pts"
+                android:textColor="#f5a623"
+                android:textSize="11sp"
+                android:textStyle="bold"
+                android:gravity="center" />
+
+        </LinearLayout>
+
+        <!-- Divider -->
+        <View
+            android:layout_width="match_parent"
+            android:layout_height="1dp"
+            android:background="#3a4258"
+            android:layout_marginBottom="4dp" />
+
+        <!-- Team rows container -->
+        <LinearLayout
+            android:id="@+id/ll_teams_container"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:orientation="vertical" />
+
+    </LinearLayout>
+</androidx.cardview.widget.CardView>

--- a/app/src/main/res/layout/item_group_team_row.xml
+++ b/app/src/main/res/layout/item_group_team_row.xml
@@ -1,0 +1,90 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="36dp"
+    android:orientation="horizontal"
+    android:gravity="center_vertical">
+
+    <ImageView
+        android:id="@+id/iv_flag"
+        android:layout_width="24dp"
+        android:layout_height="16dp"
+        android:scaleType="fitXY" />
+
+    <TextView
+        android:id="@+id/tv_name"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_weight="1"
+        android:layout_marginStart="8dp"
+        android:textColor="#d0d8e8"
+        android:textSize="13sp"
+        android:singleLine="true"
+        android:ellipsize="end" />
+
+    <TextView
+        android:id="@+id/tv_mp"
+        android:layout_width="28dp"
+        android:layout_height="wrap_content"
+        android:textColor="#c0c8d8"
+        android:textSize="13sp"
+        android:gravity="center" />
+
+    <TextView
+        android:id="@+id/tv_w"
+        android:layout_width="28dp"
+        android:layout_height="wrap_content"
+        android:textColor="#c0c8d8"
+        android:textSize="13sp"
+        android:gravity="center" />
+
+    <TextView
+        android:id="@+id/tv_d"
+        android:layout_width="28dp"
+        android:layout_height="wrap_content"
+        android:textColor="#c0c8d8"
+        android:textSize="13sp"
+        android:gravity="center" />
+
+    <TextView
+        android:id="@+id/tv_l"
+        android:layout_width="28dp"
+        android:layout_height="wrap_content"
+        android:textColor="#c0c8d8"
+        android:textSize="13sp"
+        android:gravity="center" />
+
+    <TextView
+        android:id="@+id/tv_gf"
+        android:layout_width="28dp"
+        android:layout_height="wrap_content"
+        android:textColor="#c0c8d8"
+        android:textSize="13sp"
+        android:gravity="center" />
+
+    <TextView
+        android:id="@+id/tv_ga"
+        android:layout_width="28dp"
+        android:layout_height="wrap_content"
+        android:textColor="#c0c8d8"
+        android:textSize="13sp"
+        android:gravity="center" />
+
+    <TextView
+        android:id="@+id/tv_gd"
+        android:layout_width="28dp"
+        android:layout_height="wrap_content"
+        android:textColor="#c0c8d8"
+        android:textSize="13sp"
+        android:gravity="center" />
+
+    <TextView
+        android:id="@+id/tv_pts"
+        android:layout_width="32dp"
+        android:layout_height="wrap_content"
+        android:textColor="#f5a623"
+        android:textSize="13sp"
+        android:textStyle="bold"
+        android:gravity="center" />
+
+</LinearLayout>

--- a/app/src/main/res/values/strings.xml
+++ b/app/src/main/res/values/strings.xml
@@ -1,3 +1,3 @@
 <resources>
-    <string name="app_name">TournamentBrackets</string>
+    <string name="app_name">WC 2026 Brackets</string>
 </resources>

--- a/app/src/main/res/values/strings.xml
+++ b/app/src/main/res/values/strings.xml
@@ -1,3 +1,14 @@
 <resources>
     <string name="app_name">WC 2026 Brackets</string>
+    <string name="home_subtitle">FIFA World Cup 2026</string>
+    <string name="home_group_standings">Group Standings</string>
+    <string name="home_brackets">Knockout Brackets</string>
+    <string name="col_mp">MP</string>
+    <string name="col_w">W</string>
+    <string name="col_d">D</string>
+    <string name="col_l">L</string>
+    <string name="col_gf">GF</string>
+    <string name="col_ga">GA</string>
+    <string name="col_gd">GD</string>
+    <string name="col_pts">PTS</string>
 </resources>

--- a/build.gradle
+++ b/build.gradle
@@ -6,7 +6,7 @@ buildscript {
         mavenCentral()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:7.3.1'
+        classpath 'com.android.tools.build:gradle:8.1.4'
     }
 }
 

--- a/gradle.properties
+++ b/gradle.properties
@@ -2,4 +2,5 @@
 android.useAndroidX=true
 android.enableJetifier=true
 org.gradle.jvmargs=-Xmx2048m -Dfile.encoding=UTF-8
+org.gradle.java.home=/Library/Java/JavaVirtualMachines/zulu-17.jdk/Contents/Home
 android.nonTransitiveRClass=true

